dealer.go

```diff
@@ -4,12 +4,19 @@ import (
 	"fmt"
 	"sync"
 
+	"github.com/hashicorp/go-immutable-radix"
```

Comment:
Please use the latest version https://github.com/hashicorp/go-immutable-radix?tab=readme-ov-file#v2